srcCpp/Infrastructure_pro.cxx

```diff
@@ -501,21 +501,118 @@ bool configureShmemTransport(
         DDS_DomainParticipantQos& qos,
         ParameterManager *_PM)
 {
-    // Number of messages that can be buffered in the receive queue.
-    int received_message_count_max = 1024 * 1024 * 2
-            / (int) _PM->get<unsigned long long>("dataLen");
-    if (received_message_count_max < 1) {
-        received_message_count_max = 1;
-    }
+    DDS_Property_t *parentProp = DDSPropertyQosPolicyHelper::lookup_property(qos.property,
+                "dds.transport.shmem.builtin.parent.message_size_max");
+    int parent_msg_size_max = atoi(parentProp->value);
+    std::cout << "message_size_max: " << parent_msg_size_max << std::endl;
+
+    /**
+     * The maximum size of a SHMEM segment highly depends on the platform.
+     * So that, we need to find out the maximum allocable space to avoid
+     * runtime errors.
+     *
+     * Perform an incremental search of allocable space in range (minSize, maxSize).
+     */
+    RTIOsapiSharedMemorySegmentHandle handle;
+    RTI_UINT64 pid = RTIOsapiProcess_getId();
+    RTIBool success = RTI_FALSE;
+    int retcode;
+    int key = rand();
+    int minSize = parent_msg_size_max;
```

Comment:
It's more intuitive if you have the format for both of the variables on the range
`minBufferSize` and `maxBufferSize`

Reply:
Done